File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        vector<int> ans;
        for(int x = 0; x < nums.size(); x++){
            int num = 1;
            for(int i = 0; i < nums.size(); i++){
                if(i == x) continue;
                num *= nums[i];
            }
            ans.push_back(num);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        def dfs(numsRemaning, ss):
            # print(ss)
            if len(ss) >= len(nums):
                res.append(ss.copy())
                return
            for i, num in enumerate(numsRemaning):
                ss.append(num)
                temp_remaning = numsRemaning[:i] + numsRemaning[i+1:]
                dfs(temp_remaning, ss)
                ss.pop()
                
        dfs(nums.copy(), [])
        return res